model_inspector/inspect/any_model.py

```diff
@@ -28,11 +28,22 @@ class _Inspector:
     subclasses directly.
     """
 
-    def __init__(self, model: BaseEstimator, X: pd.DataFrame, y: pd.Series):
+    def __init__(
+        self,
+        # We aim to support estimators with `scikit-learn`-compatible
+        # interfaces from popular libraries such as `xgboost` and
+        # `catboost` even if they do not inherit from `BaseEstimator`,
+        # but I do not know of a simple way to include them in this type
+        # hint without adding those libraries as dependencies, which I
+        # would like to avoid if possible.
+        model: BaseEstimator,
+        X: pd.DataFrame,
+        y: pd.Series,
+    ):
         check_is_fitted(model)
         check_X_y(X, y)
-        if not isinstance(model, (DummyClassifier, DummyRegressor)):
-            model._check_n_features(X, reset=False)
```

Comment:
`catboost`'s estimators don't have `_check_n_features`. In general we should avoid depending on sklearn internals.

Reply:
👍